perf/prometheus-install/values.yaml

```diff
@@ -0,0 +1,322 @@
+fullnameOverride: "istio"
+alertmanager:
+  enabled: false
+
+grafana:
+  enabled: false
+
+kubeApiServer:
+  enabled: false
+
+kubeControllerManager:
+  enabled: false
+
+coreDns:
+  enabled: false
+
+kubeEtcd:
+  enabled: false
+
+kubeScheduler:
+  enabled: false
+
+prometheus:
+  rbac:
+    roleNamespaces: []
+  prometheusSpec:
+    scrapeInterval: "30s"
+    storageSpec: 
+      volumeClaimTemplate:
+        spec:
+          accessModes: ["ReadWriteOnce"]
+          storageClassName: ssd
+          resources:
+            requests:
+              storage: 50Gi
+        selector: {}
+    additionalScrapeConfigs:
```

Comment:
Is it possible to inject Prometheus.aaditionalsrcapecinfig from the standard Prometheus install?

I think this is avoidable duplication

Reply:
I don't think there is any trivial/built in helm support that we could use for this. We could extract the current config from kubectl, but won't work if they don't have Prometheus installed already, and we disable Prometheus here. We could pull the Istio Helm files, but it has templating stuff - I don't think you can run helm template on a single file so we would have to run helm template on the full Istio install and extract the scrape config I think.

Do you have any other ideas?

Reply:
see if you can use --set-file option and pickup the file directly from istio install.
```
 helm template 
    --set-file .Values.prometheus.prometheusSpec.additionalScrapeConfigs="${ISTIO_INSTALL}/prometheus-scrape-configs.yaml
```

Reply:
This is how I was doing things initially, but there's a problem.  Set file imports the contents of the file as a scalar string, rather than an object.  This causes the resulting configmap to fail to be parsed by prometheus' config manager, and none of the jobs make it into prometheus as a result.  I have been unable to find a workaround to this, and have considered opening an issue with helm to address the different behavior for seemingly equivalent flags...

Reply:
Maybe we have something like 
```
additionalScrapeConfigs:
  {{ toYaml .Values.scrape }}
```
Then `--set-file scrape="..."`. I think it should work... I'll test it out later today

Reply:
Helm doesn't allow template commands in values.yaml files... https://github.com/helm/helm/issues/2492

Reply:
Oh, right, forgot this was in values. Another issue is the scrape config is part of a config map that is helm templated - probably need to extract it with `kubectl get configmap prometheus`